src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -134,19 +144,71 @@ public void setPeriod(String period) {
     public void visitJavadocToken(DetailNode ast) {
         String firstSentence = getFirstSentence(ast);
         final int endOfSentence = firstSentence.lastIndexOf(period);
-        if (endOfSentence == -1) {
-            if (!firstSentence.trim().startsWith("{@inheritDoc}")) {
+        boolean temp = true;
+        if (period.equals(PERIOD)) {
```

Comment:
Why are we checking these 2?
If user supplied custom `period` we should be using that as the period and nothing else.